iBudget/utils/aws_helper.py

```diff
@@ -0,0 +1,66 @@
+"""
+File Handler
+Provides methods for upload and download processes of the object to Amazon S3 service
+"""
+import boto3
+
+from ibudget.settings import AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, \
+    AWS_STORAGE_BUCKET_NAME
+
+BOTO_S3 = boto3.resource(
+    's3',
+    aws_access_key_id=AWS_ACCESS_KEY_ID,
+    aws_secret_access_key=AWS_SECRET_ACCESS_KEY
+)
+
+BUCKET_IMAGE = BOTO_S3.Bucket(AWS_STORAGE_BUCKET_NAME)
+
+CLIENT = boto3.client(
+    's3',
+    aws_access_key_id=AWS_ACCESS_KEY_ID,
+    aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
+)
+
+def check_if_in_bucket(file):
+    """The function which accepts a file name and checks if it exists in a bucket;
+    returns boolean value
+    """
+    file = str(file)
+    bucket_list = CLIENT.list_objects_v2(Bucket=AWS_STORAGE_BUCKET_NAME)
+    for obj in bucket_list['Contents']:
+        if obj['Key'] == file:
+            return False
+    return True
+
+
+
+def upload(file):
+    """The function which takes a file name and uploads it to the bucket
+    """
+    response = BUCKET_IMAGE.put_object(Key=file.name, Body=file.read())
+    return response.key
+
+def get_image_url(key):
+    """The function which takes a file name(key), generates the URL of this file
+    and returns it
+    """
+    url = CLIENT.generate_presigned_url(
+        ClientMethod='get_object',
+        Params={
+            'Bucket': AWS_STORAGE_BUCKET_NAME,
+            'Key': key
+        }
+    )
+    for char in url:
+        if char == '?':
```

Comment:
What does it do?

Reply:
here, the boto client's method generate_presigned_url returns the url of the image and also some other information after '?', so the loop above will cut the unnecessary information and save only the clean URL itself.